x-pack/plugins/upgrade_assistant/server/routes/deprecation_logging.ts

```diff
@@ -63,4 +64,49 @@ export function registerDeprecationLoggingRoutes({ router }: RouteDependencies)
       }
     )
   );
+
+  router.get(
+    {
+      path: `${API_BASE_PATH}/deprecation_logging/count`,
+      validate: {
+        query: schema.object({
+          from: schema.string(),
+        }),
+      },
+    },
+    versionCheckHandlerWrapper(
+      async (
+        {
+          core: {
+            elasticsearch: { client },
+          },
+        },
+        request,
+        response
+      ) => {
+        const { body: indexExists } = await client.asCurrentUser.indices.exists({
+          index: DEPRECATION_LOGS_INDEX_PATTERN,
+        });
+
+        if (!indexExists) {
+          return response.ok({ body: { count: 0 } });
```

Comment:
Do we need to log or surface the error in anyway? Or will this be more of a non-issue if https://github.com/elastic/elasticsearch/issues/76802 is implemented?

Reply:
The count method fails when the index doesn't exist, so I think it's safe to check for that and return 0 without bubbling up an error. The only case in which this can occur is if the index was deliberately deleted by the user, and that would mean that there are no logs so showing 0 would still be accurate.